app/models/source.rb

```diff
@@ -1,15 +1,19 @@
 class Source < ApplicationRecord
-  validates_lengths_from_database
   has_many :reports, :through => :logs
   has_many :logs
+  validates_lengths_from_database
   validates :value, :digest, :presence => true
```

Comment:
This matches the Message class so they are exactly the same.

Reply:
they should both be:
```suggestion
  validates :value, :presence => true
```
I believe, we still want to make sure the value is there but the digest is gone

Reply:
Damn right thanks.